label_studio/ml_model_providers/models.py

```diff
@@ -42,3 +43,20 @@ class ModelProviderConnectionScopes(models.TextChoices):
 
     def has_permission(self, user):
         return user.active_organization == self.organization
+
+    def validate_api_key(self):
+        """
+        Checks if API key provided is valid
+
+        Return: boolean if API key is valid or not
+        """
+        if self.provider == self.ModelProviders.OPENAI:
```

Comment:
if `self.provider` is anything else, the fallthrough behavior is just to say that the key is valid; this may lead to bugs in the future. The fallthrough behavior should instead probably raise a NotImplemented or something